camera_system/camera_calibrator.py

```diff
@@ -50,45 +74,90 @@ def make_game_area_info(self, is_left_course: bool, game_save_path="game_course.
-            block_color_list.append(Color(color_id))
-            print("ブロック置き場%d:%s" % (i, Color(color_id).name))
-        # ベースサークル置き場
+            # ブロック上の領域に存在する色の種類とピクセル数を取得
+            color_uniqs, color_pixel_sum = self.__color_changer.search_color(
+                point[0],
+                point[1],
+                CameraCalibrator.__SEARCH_AREA_XSIZE,
+                CameraCalibrator.__SEARCH_AREA_YSIZE)
+            # 色の種類とピクセル数を配列に格納
+            # 第2引数はindexと色IDを合わせるために-1
+            # 第3引数は色を求める際に領域に対する割合で比較できるように各色のピクセル数÷全体のピクセル数
+            np.put(self.count_point[i], color_uniqs-1,
+                   color_pixel_sum/area_pixel_sum)
+        # 認識したブロックの数を把握するための配列
+        color_count = np.zeros(CameraCalibrator.__VALIDITY_COLOR_NUM)  # (赤、黄、緑、青)
+        # 各ブロックの領域に対する色の割合が高い順に色IDを割り振る
+        for i in range(block_color_list.shape[0]):
+            # 配列の最大値のインデックスを取得
+            max_index = np.unravel_index(np.argmax(self.count_point), self.count_point.shape)
+            # ブロックに対する色IDを格納する
+            block_color_list[max_index[0]] = max_index[1]+1  # indexと色IDを合わせるために+1
+            # 認識した色をカウント
+            color_count[max_index[1]] += 1
+            # 2回認識した色を候補から外す(優先順位を小さくする)
+            if color_count[max_index[1]] == 2:
+                self.count_point[:, max_index[1]] = -1
+            # 色の判別が終わったブロックを候補から外す
+            self.count_point[max_index[0], :] = -1
+
+        # ベースサークル上のブロックの色IDを求める
         for i, base in enumerate(self.__coord.base_circle):
-            # 最頻値を求めてブロックの色を判定
-            color_id = self.__color_changer.calculate_mode_color(base[0],
-                                                                 base[1],
-                                                                 CameraCalibrator.MODE_AREA_XSIZE,
-                                                                 CameraCalibrator.MODE_AREA_YSIZE)
-            base_color_list.append(Color(color_id))
-            print("ベースサークル置き場%d:%s" % (i, Color(color_id).name))
-        # 端点サークル置き場
-        # 最頻値を求めてブロックの色を判定
-        color_id = self.__color_changer.calculate_mode_color(
-            self.__coord.end_point[0][0], self.__coord.end_point[0][1],
-            CameraCalibrator.MODE_AREA_XSIZE, CameraCalibrator.MODE_AREA_YSIZE)
-        bonus_color.append(Color(color_id))
-        print("ボーナスブロック置き場%d:%s" % (i, Color(color_id).name))
+            # ブロック上の領域に存在する色の種類とピクセル数を取得
+            color_uniqs, color_pixel_sum = self.__color_changer.search_color(
+                base[0],
+                base[1],
+                CameraCalibrator.__SEARCH_AREA_XSIZE,
+                CameraCalibrator.__SEARCH_AREA_YSIZE)
+            # 色の種類とピクセル数を配列に格納
+            np.put(self.count_base[i], color_uniqs-1,
+                   color_pixel_sum/area_pixel_sum)
+        # 各ブロックの領域に対する色の割合が高い順に色IDを割り振る
+        for i in range(base_color_list.shape[0]):
+            # 配列の最大値のインデックスを取得
```

Comment:
```suggestion
            # 配列の最大値のインデックス(ブロックのインデックス, 色のインデックス)を取得
```
あってるかわからんけど、何が入っているか書いてほしい